### ISSUE TITLE
conversion to mol and `upreferred`

### ISSUE THREAD

**Issue by briochemc** [rafaqz/UnitfulMoles.jl#20]
Would it make sense to:
- be able to convert moles of anything to just moles, as in, e.g., `molC` into `mol`? (Conversion to `mol` currently errors.)
- for `upreferred` to convert to moles of stuff, i.e., `molX` with no prefix, e.g., convert `mmolC` into `molC`? (Conversion to preferred units currently returns `kg`.)

MWE: 

```julia
julia> using Unitful, UnitfulMoles

julia> using Unitful: mol

julia> using UnitfulMoles: molC

julia> 1.2molC |> mol
ERROR: DimensionError: mol and molC are not dimensionally compatible.
Stacktrace:
 [1] #s81#159
   @ ~/.julia/packages/Unitful/woO6b/src/conversion.jl:12 [inlined]
 [2] var"#s81#159"(::Any, s::Any, t::Any)
   @ Unitful ./none:0
 [3] (::Core.GeneratedFunctionStub)(::Any, ::Vararg{Any})
   @ Core ./boot.jl:580
 [4] uconvert(a::Unitful.FreeUnits{(mol,), 𝐍, nothing}, x::Quantity{Float64, 𝐌, Unitful.FreeUnits{(molC,), 𝐌, nothing}})
   @ Unitful ~/.julia/packages/Unitful/woO6b/src/conversion.jl:78
 [5] FreeUnits
   @ ~/.julia/packages/Unitful/woO6b/src/types.jl:99 [inlined]
 [6] |>(x::Quantity{Float64, 𝐌, Unitful.FreeUnits{(molC,), 𝐌, nothing}}, f::Unitful.FreeUnits{(mol,), 𝐍, nothing})
   @ Base ./operators.jl:966
 [7] top-level scope
   @ REPL[4]:1

julia> 1.2molC |> upreferred
0.014413199999999998 kg
```

**Comment by briochemc** [comment]
On the same topic, the behavior is different for custom mol units (and IMHO the predefined mol units should match that of custom ones):

```julia
julia> @mol ABC
molABC

julia> 1molABC |> upreferred
1 mol

julia> 1molC |> upreferred
0.012010999999999999 kg
```

**Comment by rafaqz** [comment]
Oh that happens if you dont specify if a mass - then its convertible to 'mol'

I think the problem with conversion to mol or mass could be that Unitful will just convert them to mol automatically? 

I dont know what to do about that.

**Comment by briochemc** [comment]
Yeah I just realized that was actually the mechanism with which moles are converted to mass...

**Comment by briochemc** [comment]
I gave it a quick try but without success. What about the approach in https://github.com/sostock/UnitfulEquivalences.jl? Could this UnitfulMoles keep the upreferred as moles but include an "equivalency" for each element to convert from moles to mass?

**Comment by rafaqz** [comment]
Maybe, I dont know. Isnt the problem is will let you convert between Mol types, which isn't what you want if you have specificified that they are different things? But I done know, I havent actually used this for years.

**Comment by briochemc** [comment]
One could allow for direct conversions between `molX` and `mol`, between `kgX` and `X`, and add an equivalence for conversion from `molX` to `kgX`? Sure you could go from `molX` to `molY` indirectly, but that would require two steps (e.g., `molX |> mol |> molY`) and I'd consider that to be the user's responsibility (not to do that)! 🤷

**Comment by rafaqz** [comment]
Im not sure how equivalences work but hat sounds good

**Comment by briochemc** [comment]
AFAIK equivalences are just relationships with a constant, e.g., for carbon it would be the constant `12.0107gC/molC`.
